lib/beaker/hypervisor/vmpooler.rb

```diff
@@ -15,6 +15,8 @@ class Vmpooler < Beaker::Hypervisor
       Errno::ENETUNREACH,
     ]
 
+    attr_reader :options, :logger, :hosts, :credentials
```

Comment:
It's documented in the commit message: the main reason is that I prefer not to reach into an object's instance variables during testing if at all possible, that brings me to adding `:credentials`. 

If I'm going to do that I'd prefer to have them all there (the question "why is A an attribute but not B?" isn't usually best answered by "because we only tested A").

The other reason isn't documented, but, using accessors (`:logger`) instead of instance variables (`@logger`) even inside the class that owns them, ends up being less brittle, reduces coupling, and makes replacing access techniques easier; it also promotes easier testability.  I didn't make the move to changing those internal instance variable usages to calls to the accessors, but it wouldn't be out of hand.

The two arguments I can think of against doing this are: (1) there's something about these attributes (even as readers) that makes it a bad idea to expose them to the outside, and (2) [Liskov](http://en.wikipedia.org/wiki/Liskov_substitution_principle) might object.  The question might be: do we want to make this a uniformly exposed interface on any of the other classes in the family?